Add soroban-dev configuration for local smart contract development (#373)
stellar/quickstart: README.md

In test network mode, the node will join the network that developers use while developing software. Use the [Stellar Laboratory](https://laboratory.stellar.org/#account-creator?network=test) to create an account on the test network.

> #### `--futurenet`
> 
> In futurenet network mode, the node will join a new development network that developers use while developing smart contracts on stellar. Use the [Stellar Laboratory](https://laboratory.stellar.org/#account-creator?network=test) to create an account on the futurenet network.
> 

Reviewer: ```suggestion
#### `--futurenet`

**Note that FutureNet is not yet deployed.**

In futurenet network mode, the node will join a new development network that developers use while developing smart contracts on stellar. Use the [Stellar Laboratory](https://laboratory.stellar.org/#account-creator?network=test) to create an account on the futurenet network.

```

In test network mode, the node will join the network that developers use while developing software. Use the [Stellar Laboratory](https://laboratory.stellar.org/#account-creator?network=test) to create an account on the test network.

> #### `--futurenet`
> 
> **Note that FutureNet is not yet deployed.**
> 
> In futurenet network mode, the node will join a new development network that developers use while developing smart contracts on stellar. Use the [Stellar Laboratory](https://laboratory.stellar.org/#account-creator?network=test) to create an account on the futurenet network.

Reviewer: ```suggestion
#### `--futurenet`

In futurenet network mode, the node will join the [Soroban] test network that developers use while developing smart contracts on Stellar.

[Soroban]: https://soroban.stellar.org
```

*Disclaimer*: The DigitalOcean server is publicly accessible on the Internet. Do not put sensitive information on the network that you would not want someone else to know. Anyone with access to the network will be able to use the root account above.

### Soroban Dev

For local development on smart contracts on stellar(aka soroban), it is recommended to run the soroban stack locally via the `stellar/quickstart:soroban-dev` image:

```
$ docker run --rm -it \
    -p "8000:8000" \
    --name stellar \
    stellar/quickstart:soroban-dev \

>     --futurenet \

Reviewer: I think at this stage we still recommend `--standalone`?

Author: ok, I'll change, using `standalone` is part of the expected [dapp workflow](https://github.com/stellar/soroban-example-dapp#user-workflows) correct? Do we need to adjust the `standalone` core cfg in any relation to soroban support or is ok as-is?

Reviewer: Standalone core cfg should be fine as is afaik.

*Disclaimer*: The DigitalOcean server is publicly accessible on the Internet. Do not put sensitive information on the network that you would not want someone else to know. Anyone with access to the network will be able to use the root account above.

> ### Soroban Dev

Reviewer: ```suggestion
### Soroban Development
```

> For local development on smart contracts on stellar(aka soroban), it is recommended to run a `standalone` network and the soroban stack locally via the `stellar/quickstart:soroban-dev` image:

Reviewer: ```suggestion
For local development of smart contracts on Stellar using [Soroban], run a `standalone` network and the Soroban stack locally via the `stellar/quickstart:soroban-dev` image:
```

```
$ docker run --rm -it \
    -p "8000:8000" \
    --name stellar \
    stellar/quickstart:soroban-dev \
    --standalone \
    --enable-soroban-rpc \
    --protocol-version 20
```

> This will run most recent versions of smart contract(soroban) enabled: stellar core, stellar horizon, soroban rpc server. Additionally, if you want to run even more bleeding edge versions of that stack locally, it is possible to checkout [quickstart](https://github.com/stellar/quickstart.git) repo, edit `Dockerfile.soroban-dev` change `STELLAR_CORE_VERSION` and `HORIZON_VERSION` to newer debian package versions and then run `make build-soroban-dev`

Reviewer: We don't need people to edit the files if they want to run a newer version, as they can override the `ARG` directives in the file when building.
````suggestion
This will run development versions stellar-core, horizon, friendbot, and soroban-rpc server that have Soroban enabled.

If you want to run even more bleeding edge versions of that stack locally, you can build the soroban-dev image with custom versions of stellar-core, horizon, etc:
```
docker build \
    --platform linux/amd64 \
    --no-cache \
    -t stellar/quickstart:soroban-dev \
    -f Dockerfile.soroban-dev \
    --arg STELLAR_CORE_VERSION=... \
    --arg HORIZON_VERSION=... \
    git://github.com/stellar/quickstart
```
````

Author: oh nice, build from remote gh repo on the fly.

*Disclaimer*: The DigitalOcean server is publicly accessible on the Internet. Do not put sensitive information on the network that you would not want someone else to know. Anyone with access to the network will be able to use the root account above.

### Soroban Development

For local development of smart contracts on Stellar using [Soroban], run a `standalone` network and the Soroban stack locally via the `stellar/quickstart:soroban-dev` image:

```
$ docker run --rm -it \
    -p "8000:8000" \
    --name stellar \
    stellar/quickstart:soroban-dev \
    --standalone \
    --enable-soroban-rpc \

>     --protocol-version 20

Reviewer: Why is protocol version 20 mentioned here? It shouldn't be required. The quickstart image will automatically upgrade the network to its maximum available network.

Author: it gave some visibility of the network level at use for soroban, removed it, convention over config,
https://github.com/stellar/quickstart/pull/373/commits/73c0b9fd87e1d82b3a55b5063ed1999d201a49e5
